### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by Sandoryk**
## React Boilerplate

Thank you for contributing! Please take a moment to review our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
to make the process easy and effective for everyone involved.

**Please open an issue** before embarking on any significant pull request, especially those that
add a new library or change existing tests, otherwise you risk spending a lot of time working
on something that might not end up being merged into the project.

Before opening a pull request, please ensure:

- [ ] You have followed our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
- [ ] double-check your branch is based on `dev` and targets `dev` 
- [ ] Pull request has tests (we are going for 100% coverage!)
- [ ] Code is well-commented, linted and follows project conventions
- [ ] Documentation is updated (if necessary)
- [ ] Internal code generators and templates are updated (if necessary)
- [ ] Description explains the issue/use-case resolved and auto-closes related issues

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

**IMPORTANT**: By submitting a patch, you agree to allow the project
owners to license your work under the terms of the [MIT License](https://github.com/react-boilerplate/react-boilerplate/blob/master/LICENSE.md).

### CODE REVIEW

File: app/containers/App/actions.js [hunk 1]
```diff
@@ -61,3 +63,17 @@ export function repoLoadingError(error) {
     error,
   };
 }
+
+export function setSearchFilterMode(searchBy) {
+  return {
+    type: SET_SEARCH_FILTER,
+    filter: searchBy,
```
Comment (Hliebov) [inline app/containers/App/actions.js]: better name filter to "payload"

File: app/containers/App/actions.js [hunk 2]
```diff
@@ -61,3 +63,17 @@ export function repoLoadingError(error) {
...
+  };
+}
+
+export function setSortingMode(sortBy) {
+  return {
+    type: SET_SORTING,
+    sorting: sortBy,
```
Comment (Hliebov) [inline app/containers/App/actions.js]: sorting is "payload" as well

File: app/containers/SearchMode/index.js [hunk 3]
```diff
@@ -1,36 +1,51 @@
...
-        let mode = props.value == undefined || props.value == true ? true: false;
-        this.state = {
-            titleMode: mode,
-            focusColor: 'button-background-color'
-        };
-    }
+class SearchMode extends React.Component { // eslint-disable-line react/prefer-stateless-function
+  constructor(props) {
+    super(props);
+    this.focusColor = 'button-background-color';
```
Comment (Hliebov) [inline app/containers/SearchMode/index.js]: move focusColor to public property (no need to assign it in constructor)

File: app/containers/SortPanel/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import Label from 'components/Label';
+import Caption from 'components/Caption';
+import Button from 'components/Button';
+import { sortBy } from '../../stateValues'
+import { setSortingMode } from '../App/actions';
+
+class SortPanel extends React.Component { // eslint-disable-line react/prefer-stateless-function
+    constructor(props) {
+        super(props); 
+        this.focusColor = 'button-background-color';
```
Comment (Hliebov) [inline app/containers/SortPanel/index.js]: same here

File: app/containers/SortPanel/index.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+const mapStateToProps = state => {
+    return {
+        sortByMode: state.getIn(['global', 'sortedBy']),
+        resultsCount: state.getIn(['global', 'resultsCount']),
+    }
+}
+
+const mapDispatchToProps = dispatch => {
+    return {
+        setSorting: sortBy => { dispatch(setSortingMode(sortBy)) }
```
Comment (Hliebov) [inline app/containers/SortPanel/index.js]: no need for curly braces if writing in one line

File: app/containers/App/actions.js (hunk 1 above)
Comment (Sandoryk) [reply app/containers/App/actions.js]: Done

File: app/containers/App/actions.js (hunk 2 above)
Comment (Sandoryk) [reply app/containers/App/actions.js]: Done

File: app/containers/SearchMode/index.js (hunk 3 above)
Comment (Sandoryk) [reply app/containers/SearchMode/index.js]: Done

File: app/containers/SortPanel/index.js (hunk 4 above)
Comment (Sandoryk) [reply app/containers/SortPanel/index.js]: Done

File: app/containers/SortPanel/index.js (hunk 5 above)
Comment (Sandoryk) [reply app/containers/SortPanel/index.js]: Done

**Pull request merged**